shell/components/xen/xen-state.js

```diff
@@ -49,58 +49,70 @@ export default Base => class extends Base {
     Object.assign(this._state, state);
     this._invalidate();
   }
+  _setIfDirty(object) {
```

Comment:
Document briefly in a comment what this does and when one might want to use it?

Reply:
I don't put comments like that in Xen. It will have to be documented elsewhere.